src/modules/esl-event-listener/README.md

```diff
@@ -404,6 +427,130 @@ ESLEventUtils.initDescriptor(TestCases.prototype, 'onEventManualDesc', {event: '
 
 ---
 
+## <a name="extended-event-targets">Extended `EventTarget`s and standard optimizations</a> <i class="badge badge-sup badge-warning">beta</i>
+
+<a name="-esleventutilsdecorate"></a>
+
+### ⚡ `ESLEventUtils.decorate` and `ESLEventTargetDecorator`
+
+In case the original event of the target happens too frequently to be handled every time,
+it might be useful to limit its processing. In purpose to do that ESL allows creating decorated`EventTargets`.
+The decorated target will process the original target events dispatching with the passed async call decoration function
+(such as debounce or throttle).
+
+The `ESLEventUtils.decorate` creates an instance of `ESLEventTargetDecorator` that decorates
+passed original `EventTarget` event emitting. The instances of `ESLEventTargetDecorator` are lazy
+and do not subscribe to the original event until they have their own subscriptions of the same event type.
+
+⚠ Note `ESLEventUtils.decorate` method is cached, so created instances will be reused if the inner cache does not
+refuse additional arguments of the decorator. The cache does not handle multiple and non-primitive arguments.
+
+```typescript
+ESLEventUtils.decorate(
+  target: EventTarget,
+  decorator: (fn: EventListener, ...args: any[]) => EventListener,
+  ...args: any[]
+): ESLEventTargetDecorator;
+```
+
+**Parameters**:
+
+- `target` - original `EventTarget` to consume events;
+- `decorator` - decoration function to decorate original target `EventListener`s;
+- `args` - optional arguments to pass to `decorator`.
+
+#### Sharing of the decorated targets
+
+As was mentioned above, method `ESLEventUtils.decorate` (alias for `ESLEventTargetDecorator.cached`) comes with
+out of a cache for simple cases. But in some cases, we might be interested in creating wrappers with a complex
+param or we want to limit params usage across the project.
+
+It might sound obvious, but there is no any restrictions to sharing exact instances instead of using method cache.
+
+```typescript
+// shared-event-targets.ts
+export const DEBOUNCED_WINDOW = ESLEventUtils.decorate(window, debounce, 1000);
+```
+
+```typescript
+// module.ts
+class Component {
+  @listen({event: 'resize', target: DEBOUNCED_WINDOW})
+  onResize() {}
+}
+```
+
+#### Optimize `window.resize` handling with debouncing
+
+```typescript
+import {debounce} from '.../debounce';
+
+ESLEventUtils.subscribe(host, {
+  event: 'resize',
+  target: /* instead just window */ ESLEventUtils.decorate(window, debounce, 250)
+}, onResizeDebounced);
+```
+
+The sample above allows you to reuse debounced by 250 seconds version of the window,
+to receive fewer `resize` events
+(same as any other event types observed on debounced window version)
+
+#### Optimize `window.scroll` handling with throttling
+
+```typescript
+import {throttle} from '.../throttle';
+
+ESLEventUtils.subscribe(host, {
+  event: 'scroll',
+  target: /* instead just window */ ESLEventUtils.decorate(window, throttle, 250)
```

Comment:
```suggestion
  target: /* instead just window */ ESLEventUtils.decorate(window, throttle, 250)
```
```suggestion
  target: /* instead of window */ ESLEventUtils.decorate(window, throttle, 250)
```